### ISSUE TITLE
Sender aliases

### ISSUE THREAD

**Issue by tflori** [postfixadmin/postfixadmin#42]
It will be good to allow users different senders. For example I have an alias from `name@domain1.tld` to `name@domain2.tld`. The user `name@domain2.tld` should be able to send emails from email address `name@domain1.tld`.

How I solved it:
I created a table and changed `sql/sender-login-maps.cf`. It would be nice to configure this table via postfixadmin.

The table:

```sql
CREATE TABLE sender_alias (
  username VARCHAR(255) NOT NULL,
  sender VARCHAR(255) NOT NULL,
  PRIMARY KEY (username, sender)
)
```

The new query for `sql/sender-login-maps.cf`:

```sql
SELECT sender FROM (SELECT s.username, sender FROM sender_alias as s JOIN mailbox as m ON m.username = s.username AND m.active = 1 UNION SELECT username, username as sender FROM mailbox WHERE active = 1) AS stmt WHERE username = '%s'
```

If the user don't need sender aliases this table does not have to be filled for him.

**Comment by tflori** [comment]
argh - sorry didn't test it before. It has to find usernames that are vaild. So the query will be:

```sql
SELECT username FROM (SELECT s.username, sender FROM sender_alias as s JOIN mailbox as m ON m.username = s.username AND m.active = 1 UNION SELECT username, username as sender FROM mailbox WHERE active = 1) AS stmt WHERE sender = '%s'
```

**Comment by dimejo** [comment]
I haven't tested this, but I think you can simply use the 'default' queries in `smtpd_sender_login_maps`.

**main.cf**
```
smtpd_sender_login_maps = proxy:mysql:${sql_directory}}/mysql_virtual_alias_maps.cf,
    proxy:mysql:${sql_directory}}/mysql_virtual_alias_domain_maps.cf
```

**Comment by gnanet** [comment]
Question 1: are sender aliases also receiver aliases too?
Question 2: If Q1 is no, then some UI extension to edit mailbox would be great too

**Comment by dimejo** [comment]
> Question 1: are sender aliases also receiver aliases too?

I guess most of the time they are, but not necessarily.

**Comment by tflori** [comment]
@dimejo yes, might work too. But it will not use the table that I created. I wanted to use both: the aliases and a sender alias table

I also like the idea of having it configurable via the interface... 

_Sent from my OnePlus ONEPLUS A3003 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_
